src/graphs/lca.rs

```diff
@@ -72,6 +79,6 @@ impl LCA {
         if le > ri {
             std::mem::swap(&mut le, &mut ri);
         }
-        self.p[self.rmq.query(le + 1..ri + 1).1]
+        self.p[self.rmq.query(le + 1..ri + 1).1].unwrap()
```

Comment:
here, I prefer to use option as it's extra testing:

it asserts that the result of the rmq query is never the root

Reply:
valid